features/zoek-met-postcode-huisnummer.feature

```diff
@@ -0,0 +1,28 @@
+#language: nl
+
+Functionaliteit: Zoek met postcode en huisnummer
+
+Rule: Postcode en huisnummer zijn verplichte parameters
+
+    Scenario: Zoek met valide postcode en huisnummer
+
+        Gegeven het systeem heeft een persoon met de volgende gegevens
+        | naam                | waarde    |
+        | burgerservicenummer | 999995078 |
+        En de persoon heeft de volgende verblijfplaats gegevens
+        | naam       | waarde |
+        | postcode   | 2628HJ |
+        | huisnummer | 2      |
+        Als personen wordt gezocht met de volgende parameters
+        | naam       | waarde                      |
+        | type       | ZoekMetPostcodeEnHuisnummer |
+        | postcode   | 2628HJ                      |
+        | huisnummer | 2                           |
+        | fields     | burgerservicenummer         |
+        Dan bevat de response alleen personen met de volgende gegevens 
+        | burgerservicenummer |
+        | 999995078           |
+
+Scenario: Zoek met valide postcode, huisnummer, voornamen, geboortedatum, geboorteplaats, geslachtsaanduiding, gemeenteVanInschrijving, huisletter, huisnummertoevoeging
+
+Scenario: Zoek met valide postcode, huisnummer, gemeentevaninschrijving, naamopenbareruimte(?) en identificatienummeraanduiding
```

Comment:
dit scenario voegt naar mijn mening weinig toe. Ik stel voor om dit niet uit te werken, maar te laten vervallen.

Reply:
behalve wanneer je wilt laten zien dat bepaalde parameters niet zijn toegestaan in deze combinatie. En hoe dat getoond wordt. Bijvoorbeeld met invalidParams
```
{
    "invalidParams": [
        {
            "name": "naamopenbareruimte",
            "code": "unknownParam",
            "reason": "Parameter is niet verwacht."
        },
        {
            "name": "identificatienummeraanduiding",
            "code": "unknownParam",
            "reason": "Parameter is niet verwacht."
        }
    ],
    "type": "https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?#System_Net_HttpStatusCode_BadRequest",
    "title": "Een of meerdere parameters zijn niet correct.",
    "status": 400,
    "detail": "De foutieve parameter(s) zijn: naamopenbareruimte,identificatienummeraanduiding.",
    "instance": "https://www.haalcentraal.nl/haalcentraal/api/brp/ingeschrevenpersonen?burgerservicenummer=999999321&bestaatniet=parameter%20die%20niet%20bestaat",
    "code": "paramsValidation"
}
```

Of met "unsupportedCombi" (zoals in de BAG API, alleen dan liever met een fatsoenlijke toelichting in detail):
{
    "type": "https://tools.ietf.org/html/rfc7231#section-6.5.1",
    "title": "De combinatie van opgegeven parameters is niet toegestaan.",
    "status": 400,
    "detail": "Bad request.",
    "instance": "https://api.bag.acceptatie.kadaster.nl/esd/huidigebevragingen/v1/adresseerbareobjecten?nummeraanduidingIdentificatie=0599200000193766&pandIdentificatie=0599100000103376",
    "code": "unsupportedCombi"
}

Reply:
Persoonlijk denk ik dat het een beter design is als je combinaties die niet mogelijk zijn/wil maken ook niet opneem in het contract. Dan kan de consumer de fout niet maken en is het niet nodig om validatie hiervoor te implementeren

Reply:
> Persoonlijk denk ik dat het een beter design is als je combinaties die niet mogelijk zijn/wil maken ook niet opneem in het contract. Dan kan de consumer de fout niet maken en is het niet nodig om validatie hiervoor te implementeren

helemaal mee eens